File: solutions/lua/resistor-color-trio/1/resistor-color-trio.lua
--[[
  we have to get the color codes outside the main function
  then, we bring the values of the first 2 colors concatonated
  then for the third value, we use it as a raise to the power of n eg, 10^n
  Bonus: if n > 2, we use kilo

]]
local codes = {
  black = 0, brown = 1, red = 2, orange = 3, yellow = 4,
  green = 5, blue = 6, violet = 7, grey = 8, white = 9
}
return {
  decode = function(c1, c2, c3)

    local first  = codes[c1]
    local second  = codes[c2]
    local third  = codes[c3]
    
    local result_string = tostring(first) .. tostring(second)
    local final_value = tonumber(result_string) * (10 ^ third)

    if final_value >= 10^9 then
      return math.floor(final_value)/10^9, "gigaohms"
    elseif final_value >= 10^6 then
      return math.floor(final_value)/10^6, "megaohms"
    elseif final_value >= 10^3  then
      return math.floor(final_value)/10^3, "kiloohms"
    else
      return math.floor(final_value), "ohms"
    end
  end
}

File: solutions/typescript/resistor-color-trio/1/resistor-color-trio.ts
export function decodedResistorValue(colors: string[]) {
  //throw new Error('Remove this line and implement the function')
  var firstColor = Colors[colors[0] as keyof typeof Colors];
  var secondColor = Colors[colors[1] as keyof typeof Colors];
  var thirdColor = Colors[colors[2] as keyof typeof Colors];
  var value = Number(`${firstColor}${secondColor}`) * (10 ** Number(thirdColor));

 //return `${value} ohms`;
  if (thirdColor > 2 && thirdColor < 6){
    var newValue = value / (10 ** 3);
    return `${newValue} kiloohms`;
  }
  else if (thirdColor > 5 && thirdColor < 9){
    var newValue = value / (10 ** 6);
    return `${newValue} megaohms`;
  }
  else if (thirdColor > 8 && thirdColor < 12){
    var newValue = value / (10 ** 9);
    return `${newValue} gigaohms`;
  }
  else if (thirdColor == 2){
    var newValue = value/ (10 ** 3);
    return `${newValue} kiloohms`;
  }
  else { return `${value} ohms`;}
}

enum Colors {
  black = 0,
  brown = 1,
  red = 2,
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9,
}